ui/src/dashboards/containers/DashboardPage.tsx

```diff
@@ -162,10 +194,18 @@ class DashboardPage extends Component<Props, State> {
   }
 
   public componentDidUpdate(prevProps: Props) {
-    const {dashboard, autoRefresh, annotationsDisplaySetting} = this.props
+    const {
+      dashboard,
+      refreshRate,
+      updateQueryParams,
+      annotationsDisplaySetting,
+    } = this.props
 
     const prevPath = getDeep(prevProps.location, 'pathname', null)
     const thisPath = getDeep(this.props.location, 'pathname', null)
+    const localStorageRefresh = autoRefreshOptions.find(
+      r => r.milliseconds === refreshRate
+    )
```

Comment:
this predicate function `r => r.milliseconds === refreshRate` is repeated a couple of times, what do you think about making it a variable and just referring to it when you call `autoRefreshOptions.find`?

```ts
autoRefreshOptions.find(refreshRatePredicate)
```

Reply:
Sure thing.